### PR TITLE
[Enhancement]  improved eslint-config to be more readable

### PR DESCRIPTION
**Pull request by Gherciu**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [x] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] New/updated tests are included
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
updated eslint-config to be more readable and more understandable

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -22,8 +25,8 @@ module.exports = {
     "WXEnvironment": true
   },
   rules: {
-    'no-console': process.env.NODE_ENV !== 'production' ? 0 : 2,
-    'no-useless-escape': 0,
-    'no-empty': 0
+    'no-console': process.env.NODE_ENV !== 'production' ? OFF : ERROR,
```
Comment (fisker) [inline .eslintrc.js]:
```suggestion
    'no-console': process.env.NODE_ENV !== 'production' ? 'off' : 'error',
```

File: .eslintrc.js [hunk 2]
```diff
@@ -22,8 +25,8 @@ module.exports = {
...
+    'no-useless-escape': OFF,
```
Comment (fisker) [inline .eslintrc.js]:
```suggestion
    'no-useless-escape': 'off',
```

File: .eslintrc.js [hunk 3]
```diff
@@ -22,8 +25,8 @@ module.exports = {
...
+    'no-empty': OFF
```
Comment (fisker) [inline .eslintrc.js]:
```suggestion
    'no-empty': 'off'
```

File: .eslintrc.js (hunk 1 above)
Comment (fisker) [reply .eslintrc.js]: I think string 'off' is better, readable and used everywhere in eslint site

File: .eslintrc.js [hunk 4]
```diff
@@ -1,3 +1,6 @@
+const OFF = 0
```
Comment (fisker) [inline .eslintrc.js]:
```suggestion
```

File: .eslintrc.js [hunk 5]
```diff
@@ -1,3 +1,6 @@
...
+const ERROR = 2
```
Comment (fisker) [inline .eslintrc.js]:
```suggestion
```

File: .eslintrc.js (hunk 1 above)
Comment (Gherciu) [reply .eslintrc.js]: I made all suggested changes in this PR #10551. (Unfortunately, I deleted the previous fork of the project and at the moment I do not have a possibility to edit this PR)

File: .eslintrc.js (hunk 2 above)
Comment (Gherciu) [reply .eslintrc.js]: I made all suggested changes in this PR #10551. (Unfortunately, I deleted the previous fork of the project and at the moment I do not have a possibility to edit this PR)

File: .eslintrc.js (hunk 3 above)
Comment (Gherciu) [reply .eslintrc.js]: I made all suggested changes in this PR #10551. (Unfortunately, I deleted the previous fork of the project and at the moment I do not have a possibility to edit this PR)

**Comment by Gherciu** [comment]
I made all suggested changes in this PR #10551. (Unfortunately, I deleted the previous fork of the project and at the moment I do not have a possibility to edit this PR)

**Pull request closed without merging**